iwyu.cc

```diff
@@ -1445,9 +1439,18 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
     return underlying_type;
   }
 
+  set<const Type*> GetProvidedTypes(const Type* type,
```

Comment:
This refactor looks nice on its own, could you pull it out into a separate PR just to reduce the scope of this one? Thanks.